### DISCUSSION TITLE
IHftStraCtx和HftStrategy是什么?

### DISCUSSION THREAD

**Discussion by chenchenwenfeng** [wondertrader/wondertrader#81] [Q&A] [answered]
看名字，IHftStraCtx是上下文，HftStrategy是策略。那么上下文IHftStraCtx为什么还需要on_tick和on_bar 之类的方法？

**Comment by ZzzzHeJ** [comment, marked as answer]
这个是设计模式中的一种设计方式，用到了IHftStraCtx是父类，HftStrategy还有CtaStrategy都是它的子类，HftStrategy需要重写父类方法来实现多态。如果想不明白的话，就先看看设计模式
